traitsui/qt4/ui_base.py

```diff
@@ -387,7 +390,7 @@ def _add_statusbar(self):
                     name = name[col + 1 :]
                 obj = self.ui.context[obj]
                 set_text = self._set_status_text(item_control)
-                obj.on_trait_change(set_text, name, dispatch="ui")
+                obj.observe(set_text, name, dispatch="ui")
```

Comment:
These need to be disconnected at some point (I think the lack of being disconnected is causing the latest CI failure), although I'm not sure where/when that should happen.  I suppose in the dispose method with a `if self.ui.view.statusbar is not None` check?

For now I tried adding a dumb fix to just check if the control exists in the handler before trying to call `.set_text()` on it